developing-packages.md

```diff
@@ -0,0 +1,174 @@
+---
+layout: page
+title: Developing Packages & Plugins
+permalink: /developing-packages/
+---
+
+* TOC
+{:toc}
+
+## Package introduction
+
+Dart packages enable the creation of modular code that can easily be shared. A
+minimal package with a single library consists of:
+
+* A `pubspec.yaml` file: A metadata file that declared the package name,
+  version, author, etc.
+
+* A `lib` directory containing the public code in the package, minimally a
+  single `<package-name>.dart` file.
+
+For additional details, see the [Dart library package](https://www.dartlang.org/guides/libraries/create-library-packages) documentation.
+
+### Package types {#types}
+
+Flutter supports several kinds of packages:
```

Comment:
Maybe the framing is better as : "there are two kinds of code you can put into a Flutter package: dart code, and native code. If you are writing Dart code that has a dependency on Flutter, do this: ...."

(instead of "different kinds of packages", let's use "packages can contain different kinds of code")